<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>GWAS catalog</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">gtex-gwas-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="miscellaneous_statistics.html">Statistics</a>
</li>
<li>
  <a href="count_detections.html">Detections</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">GWAS catalog</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>gtex-gwas-analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.3.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190312code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190312)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190312code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190312)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhakyimlabgtexgwasanalysistree4a0996580dd8791f422a7bfda12c863a3e2790fdtargetblank4a09965a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/hakyimlab/gtex-gwas-analysis/tree/4a0996580dd8791f422a7bfda12c863a3e2790fd" target="_blank">4a09965</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomhakyimlabgtexgwasanalysistree4a0996580dd8791f422a7bfda12c863a3e2790fdtargetblank4a09965a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/4a0996580dd8791f422a7bfda12c863a3e2790fd/analysis/gwas_catalog.Rmd" target="_blank">4a09965</a>
</td>
<td>
Yanyu Liang
</td>
<td>
2019-06-07
</td>
<td>
workflowr::wflow_publish(“gwas_catalog.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/18cfb9654de8f6d6ac8c202fae40081745b47de3/analysis/gwas_catalog.Rmd" target="_blank">18cfb96</a>
</td>
<td>
Yanyu Liang
</td>
<td>
2019-05-02
</td>
<td>
updated gwas catalog
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/gtex-gwas-analysis/18cfb9654de8f6d6ac8c202fae40081745b47de3/docs/gwas_catalog.html" target="_blank">18cfb96</a>
</td>
<td>
Yanyu Liang
</td>
<td>
2019-05-02
</td>
<td>
updated gwas catalog
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/gtex-gwas-analysis/d47dab0507e6bb08ae3fd20a1ee5dc0d2adc01fd/docs/gwas_catalog.html" target="_blank">d47dab0</a>
</td>
<td>
Alvaro
</td>
<td>
2019-03-26
</td>
<td>
Updated workflowr site pages.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/gtex-gwas-analysis/121dd26344d759dc31345d75c43f849c722ce3df/docs/gwas_catalog.html" target="_blank">121dd26</a>
</td>
<td>
Alvaro
</td>
<td>
2019-03-26
</td>
<td>
wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/910a320d165a74fb8b82446988f3dc9dbd3bc93f/analysis/gwas_catalog.Rmd" target="_blank">910a320</a>
</td>
<td>
Alvaro
</td>
<td>
2019-03-26
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/066a93fa0ca1f89ac98ab7289f4d13a42ae9eed8/analysis/gwas_catalog.Rmd" target="_blank">066a93f</a>
</td>
<td>
Yanyu Liang
</td>
<td>
2019-03-26
</td>
<td>
changed gwas catalog table name to make it more clear
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/0bed6901cf06042f95031661ae9940ed680eb7f0/analysis/gwas_catalog.Rmd" target="_blank">0bed690</a>
</td>
<td>
Yanyu Liang
</td>
<td>
2019-03-26
</td>
<td>
add gwas catalog tables and basic stats
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(pander)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(reshape2)
library(ggplot2)</code></pre>
<div id="analysis-pipeline" class="section level1">
<h1>Analysis pipeline</h1>
<p>First of all, we extracted all GWAS catalog variants which were identified as GTEx v8 variant (by matching rsID in lookup table). These variants were defined as <strong>GWAS catalog variants</strong> in this analysis. Next, we defined a variant to be <strong>cis-eVariant</strong> if it was identified as cis-eQTL in at least one gene and one tissue (at FDR &lt; 0.05). Similarly, we defined a variant to be <strong>cis-sVariant</strong> if it was identified as cis-sQTL in at least one cluster (only one cluster was used for each gene) and one tissue (at FDR &lt; 0.05). In the same manner, we defined a variant to be <strong>trans-eVariant</strong> if it was identified as trans-eQTL in at least one gene one tissue (at FDR &lt; 0.1). Note that <strong>cis-eVariant</strong>, <strong>cis-sVariant</strong>, and <strong>trans-eVariant</strong> were counted in <strong>signif</strong> entry in the following 2-by-2 table.</p>
<p>We obtained the list of variants that were tested for cis-eQTL and cis-sQTL by scanning through the files reporting all tested variant-gene/cluster pairs. We obtained the list of variants that were tested for trans-eQTL by scanning through the files in <code>trans-eQTL/variant_list/</code> which contained the list of variants included in trans-eQTL testing by tissue. Note that these variant lists were counted in <strong>tested</strong> entry in the following 2-by-2 table.</p>
</div>
<div id="about-loaded-data.frame" class="section level1">
<h1>About loaded data.frame</h1>
<p>The script to generate the input data.frame can be found <a href="https://bitbucket.org/yanyul/rotation-at-imlab/src/master/analysis/annotate_gwas_catalog/summary/output/gen_table_sub_new.R">here</a></p>
<p>The full script is shown below</p>
<pre><code>library(data.table)
options(datatable.fread.datatable = F)
library(dplyr)
df = readRDS(&#39;tables.rds&#39;)
df_gwas_gtex_table = df$df_gwas_gtex_table
df_gwas_table = df$df_gwas_table
summary_gwas_variant &lt;- data.frame(gwas_table_idx = 1 : nrow(df_gwas_table), gwas_id = df_gwas_table$rsID)
summary_gwas_variant$gtex_id &lt;- rep(NA, nrow(df_gwas_table))
summary_gwas_variant$gtex_id[match(df_gwas_gtex_table$rsID, df_gwas_table$rsID)] &lt;- df_gwas_gtex_table$GTExID


list_signif_eqtl = fread(&#39;../../output/ALL.signif_eqtl.GTExFDRlt0.05.txt&#39;, header = F)
list_signif_sqtl = fread(&#39;../../output/ALL.signif_eqtl.GTExSQTLFDRlt0.05.txt&#39;, header = F)

list_tested_eqtl = fread(&#39;../../output/ALL.tested_eqtl.GTExV8TestedEQTL.txt&#39;, header = F)
list_tested_sqtl = fread(&#39;../../output/ALL.tested_eqtl.GTExV8TestedSQTL.txt&#39;, header = F)

list_tested_trans_eqtl = fread(&#39;../../output/ALL.tested_trans_eqtl.GTExTestedTransEQTL.txt&#39;, header = F) 
list_signif_trans_eqtl1 = fread(&#39;/gpfs/data/gtex-group/v8/trans-eQTL/trans_egenes/trans_genes_lincRNA_FDR_0.1.tsv&#39;) 
list_signif_trans_eqtl2 = fread(&#39;/gpfs/data/gtex-group/v8/trans-eQTL/trans_egenes/trans_genes_protein_coding_FDR_0.1.tsv&#39;) 
list_signif_trans_eqtl = union(list_signif_trans_eqtl1$rsid, list_signif_trans_eqtl2$rsid)

my_summarize = function(gwas_table, tested_list, signif_list) {
  # table
  # type  tested  signif
  # gwas  t_gwas  s_gwas
  # all   t_all   s_all
  t_all = length(tested_list)
  s_all = length(signif_list)
  gwas = gwas_table %&gt;% filter(!is.na(gtex_id)) %&gt;% filter(!duplicated(gwas_id)) %&gt;% mutate(is_tested = gtex_id %in% tested_list, is_signif = gtex_id %in% signif_list)
  t_gwas = sum(gwas$is_tested)
  s_gwas = sum(gwas$is_signif)
  data.frame(type = c(&#39;gwas&#39;, &#39;all&#39;), tested = c(t_gwas, t_all), signif = c(s_gwas, s_all))
}
list_tested_eqtl = list_tested_eqtl$V1
list_signif_eqtl = list_signif_eqtl$V1
list_tested_sqtl = list_tested_sqtl$V1
list_signif_sqtl = list_signif_sqtl$V1
list_tested_trans_eqtl = list_tested_trans_eqtl$V1
eqtl = my_summarize(summary_gwas_variant, list_tested_eqtl, list_signif_eqtl)
sqtl = my_summarize(summary_gwas_variant, list_tested_sqtl, list_signif_sqtl)
trans_eqtl = my_summarize(summary_gwas_variant, list_tested_trans_eqtl, list_signif_trans_eqtl)
ngwas = sum(!duplicated(summary_gwas_variant$gwas_id))


saveRDS(list(eqtl = eqtl, sqtl = sqtl, trans_eqtl = trans_eqtl, ngwas = ngwas), &#39;table_sub_new.rds&#39;)</code></pre>
</div>
<div id="load-data.frame" class="section level1">
<h1>Load data.frame</h1>
<pre class="r"><code>df = readRDS(&#39;data/gwas_catalog_tables.rds&#39;)
eqtl = df$eqtl
sqtl = df$sqtl
trans_eqtl = df$trans_eqtl
ngwas = df$ngwas</code></pre>
</div>
<div id="count-table" class="section level1">
<h1>Count table</h1>
<pre class="r"><code>eqtl %&gt;% pander(caption = &#39;cis-eVariant&#39;)</code></pre>
<table style="width:40%;">
<caption>cis-eVariant</caption>
<colgroup>
<col width="9%" />
<col width="15%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">type</th>
<th align="center">tested</th>
<th align="center">signif</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">gwas</td>
<td align="center">44137</td>
<td align="center">27819</td>
</tr>
<tr class="even">
<td align="center">all</td>
<td align="center">10743590</td>
<td align="center">4632457</td>
</tr>
</tbody>
</table>
<pre class="r"><code>sqtl %&gt;% pander(caption = &#39;cis-sVariant&#39;)</code></pre>
<table style="width:40%;">
<caption>cis-sVariant</caption>
<colgroup>
<col width="9%" />
<col width="15%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">type</th>
<th align="center">tested</th>
<th align="center">signif</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">gwas</td>
<td align="center">44029</td>
<td align="center">16197</td>
</tr>
<tr class="even">
<td align="center">all</td>
<td align="center">10299184</td>
<td align="center">2027766</td>
</tr>
</tbody>
</table>
<pre class="r"><code>trans_eqtl %&gt;% pander(caption = &#39;trans-eVariant&#39;)</code></pre>
<table style="width:38%;">
<caption>trans-eVariant</caption>
<colgroup>
<col width="9%" />
<col width="13%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">type</th>
<th align="center">tested</th>
<th align="center">signif</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">gwas</td>
<td align="center">34406</td>
<td align="center">10</td>
</tr>
<tr class="even">
<td align="center">all</td>
<td align="center">5302059</td>
<td align="center">215</td>
</tr>
</tbody>
</table>
</div>
<div id="qtl-rate-gwas-catalog-vs.overall" class="section level1">
<h1>QTL rate: GWAS catalog vs. overall</h1>
<pre class="r"><code>rate = eqtl %&gt;% mutate(rate = signif / tested, qtl = &#39;eqtl&#39;) 
rate = rbind(rate, sqtl %&gt;% mutate(rate = signif / tested, qtl = &#39;sqtl&#39;))
rate = rbind(rate, trans_eqtl %&gt;% mutate(rate = signif / tested, qtl = &#39;trans_eqtl&#39;))
rate %&gt;% ggplot() + geom_bar(aes(x = type, y = rate, fill = type), stat = &#39;identity&#39;) + facet_wrap(~qtl, scales = &#39;free_y&#39;) + ggtitle(&#39;Significant xVariant rate&#39;)</code></pre>
<p><img src="figure/gwas_catalog.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/18cfb9654de8f6d6ac8c202fae40081745b47de3/docs/figure/gwas_catalog.Rmd/unnamed-chunk-2-1.png" target="_blank">18cfb96</a>
</td>
<td>
Yanyu Liang
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="test-rate-percentage-of-gwas-catalog-variants-are-tested-for-qtl" class="section level1">
<h1>Test rate (percentage of GWAS catalog variants are tested for QTL)</h1>
<pre class="r"><code>rate = eqtl %&gt;% filter(type == &#39;gwas&#39;) %&gt;% mutate(rate = tested / ngwas, qtl = &#39;eqtl&#39;)
rate = rbind(rate, sqtl %&gt;% filter(type == &#39;gwas&#39;) %&gt;% mutate(rate = tested / ngwas, qtl = &#39;sqtl&#39;))
rate = rbind(rate, trans_eqtl %&gt;% filter(type == &#39;gwas&#39;) %&gt;% mutate(rate = tested / ngwas, qtl = &#39;trans_eqtl&#39;))
rate %&gt;% ggplot() + geom_bar(aes(x = qtl, y = rate), stat = &#39;identity&#39;) + ggtitle(&#39;Percentage of GWAS catalog variants are tested for QTL&#39;)</code></pre>
<p><img src="figure/gwas_catalog.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/18cfb9654de8f6d6ac8c202fae40081745b47de3/docs/figure/gwas_catalog.Rmd/unnamed-chunk-3-1.png" target="_blank">18cfb96</a>
</td>
<td>
Yanyu Liang
</td>
<td>
2019-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.0 (2017-04-21)
Platform: x86_64-apple-darwin16.5.0 (64-bit)
Running under: macOS  10.13.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2.2 ggplot2_3.1.0  reshape2_1.4.3 dplyr_0.7.8   
[5] pander_0.6.3  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       compiler_3.4.0   pillar_1.3.1     git2r_0.25.2    
 [5] plyr_1.8.4       workflowr_1.3.0  bindr_0.1.1      tools_3.4.0     
 [9] digest_0.6.18    evaluate_0.13    tibble_2.0.1     gtable_0.2.0    
[13] pkgconfig_2.0.2  rlang_0.3.1      rstudioapi_0.10  yaml_2.2.0      
[17] xfun_0.6         withr_2.1.2      stringr_1.4.0    knitr_1.22      
[21] fs_1.2.7         rprojroot_1.3-2  grid_3.4.0       tidyselect_0.2.5
[25] glue_1.3.1       R6_2.4.0         rmarkdown_1.12   purrr_0.2.3     
[29] magrittr_1.5     whisker_0.3-2    backports_1.1.4  scales_1.0.0    
[33] htmltools_0.3.6  assertthat_0.2.0 colorspace_1.4-0 labeling_0.3    
[37] stringi_1.2.4    lazyeval_0.2.0   munsell_0.5.0    crayon_1.3.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
