<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Alvaro" />

<meta name="date" content="2019-08-20" />

<title>gwas_enloc_predixcan_multixcan</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">gtex-gwas-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="miscellaneous_statistics.html">Statistics</a>
</li>
<li>
  <a href="count_detections.html">Detections</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">gwas_enloc_predixcan_multixcan</h1>
<h4 class="author">Alvaro</h4>
<h4 class="date">2019-08-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-10-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>gtex-gwas-analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190312code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190312)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190312code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190312)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhakyimlabgtexgwasanalysistreea71bfe16ee302b78d07081045acbc1b61b8a9d7dtargetblanka71bfe1a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/hakyimlab/gtex-gwas-analysis/tree/a71bfe16ee302b78d07081045acbc1b61b8a9d7d" target="_blank">a71bfe1</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomhakyimlabgtexgwasanalysistreea71bfe16ee302b78d07081045acbc1b61b8a9d7dtargetblanka71bfe1a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .httr-oauth

Unstaged changes:
    Modified:   analysis/gwas_enloc_predixcan_multixcan.Rmd
    Modified:   analysis/miscellaneous_statistics_2.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/222b6429a4cb62d3b66fedde20b937394a9e693a/analysis/gwas_enloc_predixcan_multixcan.Rmd" target="_blank">222b642</a>
</td>
<td>
Alvaro
</td>
<td>
2019-10-04
</td>
<td>
Updates for first revision of manuscript
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/gtex-gwas-analysis/222b6429a4cb62d3b66fedde20b937394a9e693a/docs/gwas_enloc_predixcan_multixcan.html" target="_blank">222b642</a>
</td>
<td>
Alvaro
</td>
<td>
2019-10-04
</td>
<td>
Updates for first revision of manuscript
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/e85ed91d848450426843abf0a747aed24ea25e01/analysis/gwas_enloc_predixcan_multixcan.Rmd" target="_blank">e85ed91</a>
</td>
<td>
Alvaro
</td>
<td>
2019-10-01
</td>
<td>
upset plot for s-predixcan, enloc, gwas. Additional tables with loci counts.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/gtex-gwas-analysis/e85ed91d848450426843abf0a747aed24ea25e01/docs/gwas_enloc_predixcan_multixcan.html" target="_blank">e85ed91</a>
</td>
<td>
Alvaro
</td>
<td>
2019-10-01
</td>
<td>
upset plot for s-predixcan, enloc, gwas. Additional tables with loci counts.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/gtex-gwas-analysis/51aff2045bec9eaeac486d945b65edae4309a9c9/docs/gwas_enloc_predixcan_multixcan.html" target="_blank">51aff20</a>
</td>
<td>
Alvaro
</td>
<td>
2019-09-27
</td>
<td>
More miscellaneous summary statistics. Detection by s-predixcan/s-multixcan/enoc in eQTL and sQTL. Figure for GWAS deflation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/gtex-gwas-analysis/5004281bd184761566e1f7cedaa8af748979199b/docs/gwas_enloc_predixcan_multixcan.html" target="_blank">5004281</a>
</td>
<td>
Alvaro
</td>
<td>
2019-09-23
</td>
<td>
Updated to new results
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/048c5339974d7750b0387df8f9db5395c52a4052/analysis/gwas_enloc_predixcan_multixcan.Rmd" target="_blank">048c533</a>
</td>
<td>
Alvaro
</td>
<td>
2019-08-20
</td>
<td>
Added GWAS/ENLOC/PrediXcan/Multixcan ld region counts. Updated stats.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/gtex-gwas-analysis/048c5339974d7750b0387df8f9db5395c52a4052/docs/gwas_enloc_predixcan_multixcan.html" target="_blank">048c533</a>
</td>
<td>
Alvaro
</td>
<td>
2019-08-20
</td>
<td>
Added GWAS/ENLOC/PrediXcan/Multixcan ld region counts. Updated stats.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>GWAS subgroup Figures for the main paper</p>
<div id="import-libraries-and-source-scripts" class="section level4">
<h4>Import libraries and source scripts</h4>
<pre class="r"><code>suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(UpSetR))
suppressPackageStartupMessages(library(viridis))
suppressWarnings(source(&quot;code/helpers/_helpers.R&quot;))

SUMMARY &lt;- &quot;data/summaries/mashr_regions.txt&quot;

gwas_metadata &lt;- &quot;data/gwas_metadata.txt&quot; %&gt;% read_tsv</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_character(),
  Description = col_logical(),
  Sample_Size = col_double(),
  Binary = col_double(),
  Cases = col_double(),
  Deflation = col_double()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<pre class="r"><code>RESULT&lt;-&quot;output/g&quot;
dir.create(RESULT, showWarnings = FALSE, recursive=TRUE)
fp_ &lt;- function(p) file.path(RESULT, p)</code></pre>
</div>
</div>
<div id="read-number-gwaspredixcanenloc-significant-association-per-regiontrait" class="section level2">
<h2>read number GWAS/PrediXcan/enloc significant association per region/trait</h2>
<pre><code>           enloc_eqtl            enloc_sqtl                  gwas 
                 2303                  1223                  5385 
      smultixcan_eqtl       smultixcan_sqtl       spredixcan_eqtl 
                 4083                  3667                  3899 
      spredixcan_sqtl spredixcan_eqtl_enloc spredixcan_sqtl_enloc 
                 3345                  2125                  1135 
smultixcan_eqtl_enloc smultixcan_sqtl_enloc 
                 2152                  1156 </code></pre>
<pre><code>[1] &quot;n of significant gwas block/traits&quot;</code></pre>
<pre><code>[1] 5385</code></pre>
<pre><code>[1] &quot;n of significant gwas blocks&quot;</code></pre>
<pre><code>[1] 1167</code></pre>
<pre><code>[1] &quot;n of significant gwas traits&quot;</code></pre>
<pre><code>[1] 82</code></pre>
<p>out of 5385 GWAS significant associations (in 1167 loci across 82 traits).</p>
<p>41% (2193/5385) Loci had a significant, colocalized association via eQTL on S-MultiXcan/S-PrediXcan. 22% (1160/5385) Loci had a significant, colocalized association via sQTL on S-MultiXcan/S-PrediXcan. 47% (2519/5385) Loci had a significant, colocalized association via xQTL on S-MultiXcan/S-PrediXcan.</p>
<p><img src="figure/gwas_enloc_predixcan_multixcan.Rmd/plot%20number%20of%20expression%20significant%20loci-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot number of expression significant loci-1">
Past versions of plot number of expression significant loci-1.png
</button>
</p>
<div id="fig-plot number of expression significant loci-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/222b6429a4cb62d3b66fedde20b937394a9e693a/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot number of expression significant loci-1.png" target="_blank">222b642</a>
</td>
<td>
Alvaro
</td>
<td>
2019-10-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/5004281bd184761566e1f7cedaa8af748979199b/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot number of expression significant loci-1.png" target="_blank">5004281</a>
</td>
<td>
Alvaro
</td>
<td>
2019-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/048c5339974d7750b0387df8f9db5395c52a4052/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot number of expression significant loci-1.png" target="_blank">048c533</a>
</td>
<td>
Alvaro
</td>
<td>
2019-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
<p>72% (3899/5385) GWAS loci could be associated to changes in expression of a gene using PrediXcan (aggregated across tissues), 55% (2125/3899) of which are also colocalized (enloc colocalization probability &gt; 0.5).</p>
<p><img src="figure/gwas_enloc_predixcan_multixcan.Rmd/plot%20number%20of%20splicing%20significant%20loci-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot number of splicing significant loci-1">
Past versions of plot number of splicing significant loci-1.png
</button>
</p>
<div id="fig-plot number of splicing significant loci-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/222b6429a4cb62d3b66fedde20b937394a9e693a/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot number of splicing significant loci-1.png" target="_blank">222b642</a>
</td>
<td>
Alvaro
</td>
<td>
2019-10-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/048c5339974d7750b0387df8f9db5395c52a4052/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot number of splicing significant loci-1.png" target="_blank">048c533</a>
</td>
<td>
Alvaro
</td>
<td>
2019-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
<p>62% (3345/5385) could be associated with changes in splicing of a gene using PrediXcan, but only 34% (1135/3345) of which were colocalized.</p>
<p><img src="figure/gwas_enloc_predixcan_multixcan.Rmd/plot%20significant%20expr%20and%20splicing%20loci-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot significant expr and splicing loci-1">
Past versions of plot significant expr and splicing loci-1.png
</button>
</p>
<div id="fig-plot significant expr and splicing loci-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/222b6429a4cb62d3b66fedde20b937394a9e693a/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot significant expr and splicing loci-1.png" target="_blank">222b642</a>
</td>
<td>
Alvaro
</td>
<td>
2019-10-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/5004281bd184761566e1f7cedaa8af748979199b/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot significant expr and splicing loci-1.png" target="_blank">5004281</a>
</td>
<td>
Alvaro
</td>
<td>
2019-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/048c5339974d7750b0387df8f9db5395c52a4052/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot significant expr and splicing loci-1.png" target="_blank">048c533</a>
</td>
<td>
Alvaro
</td>
<td>
2019-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
<p><img src="figure/gwas_enloc_predixcan_multixcan.Rmd/plot%20enloc%20expr%20and%20splicing%20loci-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot enloc expr and splicing loci-1">
Past versions of plot enloc expr and splicing loci-1.png
</button>
</p>
<div id="fig-plot enloc expr and splicing loci-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/5004281bd184761566e1f7cedaa8af748979199b/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot enloc expr and splicing loci-1.png" target="_blank">5004281</a>
</td>
<td>
Alvaro
</td>
<td>
2019-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/hakyimlab/gtex-gwas-analysis/blob/048c5339974d7750b0387df8f9db5395c52a4052/docs/figure/gwas_enloc_predixcan_multixcan.Rmd/plot enloc expr and splicing loci-1.png" target="_blank">048c533</a>
</td>
<td>
Alvaro
</td>
<td>
2019-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
<p>43% (2303/5385) of GWAS associations were colocalized with an eQTL whereas 23% (1223/5385) could be colocalized with an sQTL. 73% (898/898) colocalized with an sQTL were also colocalized with an eQTL.</p>
<p>92% (2125/2303) of expression colocalized associations were also significantly associated with the predicted expression of a gene. 93% (1135/1223) of splicing colocalized associations were linked to predicted splicing events.</p>
</div>
<div id="upset-plot" class="section level1">
<h1>Upset plot</h1>
<pre><code>png 
  2 </code></pre>
<pre><code>png 
  2 </code></pre>
<div id="additional-tables" class="section level2">
<h2>Additional tables</h2>
<p>Proportion of loci with associated/colocalized genes/introns</p>
<pre><code>                 method       data count
1                  loci   splicing  5385
2                 ENLOC expression  2303
3                 ENLOC   splicing  1223
4           S-PrediXcan expression  3899
5           S-PrediXcan   splicing  3345
6 S-PrediXcan\n&amp;\nENLOC expression  2125
7 S-PrediXcan\n&amp;\nENLOC   splicing  1135
8 S-PrediXcan\n&amp;\nENLOC expression   898</code></pre>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] viridis_0.5.1     viridisLite_0.3.0 UpSetR_1.4.0      ggplot2_3.2.1    
[5] readr_1.2.0       tidyr_0.8.1.9000  dplyr_0.8.0.1    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.2         pillar_1.4.2       compiler_3.4.4    
 [4] git2r_0.25.2       plyr_1.8.4         workflowr_1.2.0   
 [7] tools_3.4.4        zeallot_0.1.0      digest_0.6.20     
[10] evaluate_0.13      tibble_2.1.3       gtable_0.2.0      
[13] pkgconfig_2.0.2    rlang_0.4.0        yaml_2.2.0        
[16] xfun_0.5           gridExtra_2.2.1    withr_2.1.2       
[19] stringr_1.4.0.9000 knitr_1.22         fs_1.3.1          
[22] vctrs_0.2.0        hms_0.5.0          rprojroot_1.3-2   
[25] grid_3.4.4         tidyselect_0.2.5   glue_1.3.1.9000   
[28] R6_2.4.0           rmarkdown_1.12     purrr_0.3.2       
[31] magrittr_1.5       whisker_0.3-2      backports_1.1.4   
[34] scales_0.5.0.9000  htmltools_0.3.6    assertthat_0.2.0  
[37] colorspace_1.3-2   labeling_0.3       stringi_1.4.4     
[40] lazyeval_0.2.0     munsell_0.4.3      crayon_1.3.4      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
