---
title: "miscellaneous_statistics_2"
author: "Alvaro"
date: "2019-08-20"
output: workflowr::wflow_html
---

## Introduction

Here follow miscellaneous numbers to be plugged in the main GTEx paper, supplements, and GWAs companion (and its supplement)

```{r, echo=FALSE, warning=FALSE, message=FALSE}
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(bigrquery))

suppressWarnings(source("code/helpers/_helpers.R"))
suppressWarnings(source("code/helpers/_helpers_big_query.R"))
suppressWarnings(source("code/helpers/_helpers_big_query_tables.R"))

options(gargle_oauth_email = TRUE)

OUTPUT<-"output"
dir.create(OUTPUT, showWarnings = FALSE, recursive=TRUE)
fp_ <- function(p) file.path(OUTPUT, p)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
gwas_metadata <- (function(){
  query <- glue::glue("SELECT Tag as phenotype, Deflation as deflation, new_abbreviation as abbreviation, Category as category
                       FROM {gwas_metadata_tbl$dataset_name}.{gwas_metadata_tbl$table_name}
                       WHERE Deflation=0")
  query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})()
pheno_whitelist <- sprintf("(%s)", toString(sprintf("'%s'", gwas_metadata$phenotype)))

gencode <- (function(){
  query <- glue::glue("SELECT *
                       FROM {gencode_all_annotation_tbl$dataset_name}.{gencode_all_annotation_tbl$table_name}
                       WHERE gene_type in ('lincRNA', 'protein_coding', 'pseudogene')")
  query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})()

introns <- (function(){
  query <- glue::glue("SELECT * FROM {intron_annotation_tbl$dataset_name}.{intron_annotation_tbl$table_name}")
  query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})()
```

# ENLOC runs

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_eqtl_eur_tuples <- (function(){
  query <- 
"SELECT COUNT(*) 
 FROM (SELECT DISTINCT molecular_qtl_trait as gene_id, phenotype, tissue
       FROM {enloc_tbl_eqtl_eur$dataset_name}.{enloc_tbl_eqtl_eur$table_name} as e
       WHERE phenotype in {pheno_whitelist})" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_eqtl_eur_tuples_colocalized <- (function(){
  query <- 
"SELECT COUNT(*) 
 FROM (SELECT DISTINCT molecular_qtl_trait as gene_id, phenotype, tissue
       FROM {enloc_tbl_eqtl_eur$dataset_name}.{enloc_tbl_eqtl_eur$table_name} as e
       WHERE phenotype in {pheno_whitelist} AND locus_rcp >0.5)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_eqtl_eur_gt_pairs <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene_id, tissue
  FROM (
    SELECT molecular_qtl_trait as gene_id, phenotype, tissue
    FROM {enloc_tbl_eqtl_eur$dataset_name}.{enloc_tbl_eqtl_eur$table_name} as e
    WHERE phenotype in {pheno_whitelist}
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_eqtl_eur_genes <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene_id
  FROM (
    SELECT molecular_qtl_trait as gene_id FROM {enloc_tbl_eqtl_eur$dataset_name}.{enloc_tbl_eqtl_eur$table_name}
    WHERE phenotype in {pheno_whitelist}
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer

n_enloc_eqtl_eur_genes_approved <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene_id
  FROM (
    SELECT molecular_qtl_trait as gene_id FROM {enloc_tbl_eqtl_eur$dataset_name}.{enloc_tbl_eqtl_eur$table_name} as e
    INNER JOIN {gencode_all_annotation_tbl$dataset_name}.{gencode_all_annotation_tbl$table_name} as a
    ON e.molecular_qtl_trait = a.gene_id
    WHERE phenotype in {pheno_whitelist} AND a.gene_type  in ('lincRNA', 'protein_coding', 'pseudogene')
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_eqtl_eur_genes_colocalized <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene_id
  FROM (
    SELECT molecular_qtl_trait as gene_id FROM {enloc_tbl_eqtl_eur$dataset_name}.{enloc_tbl_eqtl_eur$table_name}
    WHERE phenotype in {pheno_whitelist} and locus_rcp >0.5
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_sqtl_eur_tuples <- (function(){
  query <- 
"SELECT COUNT(*) 
 FROM (SELECT DISTINCT molecular_qtl_trait as intron_id, phenotype, tissue
       FROM {enloc_tbl_sqtl_eur$dataset_name}.{enloc_tbl_sqtl_eur$table_name} as e
       WHERE phenotype in {pheno_whitelist})" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_sqtl_eur_tuples_colocalized <- (function(){
  query <- 
"SELECT COUNT(*) 
 FROM (SELECT DISTINCT molecular_qtl_trait as intron_id, phenotype, tissue
       FROM {enloc_tbl_sqtl_eur$dataset_name}.{enloc_tbl_sqtl_eur$table_name} as e
       WHERE phenotype in {pheno_whitelist} AND locus_rcp > 0.5)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_sqtl_eur_gt_pairs <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT intron_id, tissue
  FROM (
    SELECT molecular_qtl_trait as intron_id, phenotype, tissue
    FROM {enloc_tbl_sqtl_eur$dataset_name}.{enloc_tbl_sqtl_eur$table_name} as e
    WHERE phenotype in {pheno_whitelist}
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_sqtl_eur_genes <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene_id
  FROM (
    SELECT molecular_qtl_trait as gene_id FROM {enloc_tbl_sqtl_eur$dataset_name}.{enloc_tbl_sqtl_eur$table_name}
    WHERE phenotype in {pheno_whitelist}
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_enloc_sqtl_eur_genes_colocalized <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene_id
  FROM (
    SELECT molecular_qtl_trait as gene_id FROM {enloc_tbl_sqtl_eur$dataset_name}.{enloc_tbl_sqtl_eur$table_name}
    WHERE phenotype in {pheno_whitelist} and locus_rcp >0.5
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

We obtained:

- `r n_enloc_eqtl_eur_tuples` `(gene, tissue, phenotype)` distinct tuples from expression on Eur individuals
- `r n_enloc_eqtl_eur_tuples_colocalized` colocalzied `(gene, tissue, phenotype)` tuples
- `r n_enloc_eqtl_eur_gt_pairs` `(gene, tissue)` distinct pairs from expression on Eur individuals
- (`r n_enloc_eqtl_eur_genes_colocalized` / `r n_enloc_eqtl_eur_genes`) colocalized genes
- `r n_enloc_sqtl_eur_tuples` `(gene, tissue, phenotype)` distinct tuples from splicing on Eur individuals
- `r n_enloc_sqtl_eur_tuples_colocalized` colocalized `(gene, tissue, phenotype)` distinct tuples from splicing on Eur individuals
- `r n_enloc_sqtl_eur_gt_pairs` `(gene, tissue)` distinct pairs from splicing on Eur individuals
- (`r n_enloc_sqtl_eur_genes_colocalized` / `r n_enloc_sqtl_eur_genes`) colocalized genes

# PrediXcan Expression

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_models_gt_pairs_eqtl_eur <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene, tissue
  FROM (
    SELECT gene, tissue
    FROM {prediction_mashr_models_extra_tbl_eqtl$dataset_name}.{prediction_mashr_models_extra_tbl_eqtl$table_name} as e
    WHERE n_snps_in_model > 0
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_models_gt_pairs_sqtl_eur <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene, tissue
  FROM (
    SELECT gene, tissue
    FROM {prediction_mashr_models_extra_tbl_sqtl$dataset_name}.{prediction_mashr_models_extra_tbl_sqtl$table_name} as e
    WHERE n_snps_in_model > 0
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_models_genes_eqtl_eur <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene
  FROM (
    SELECT gene
    FROM {prediction_mashr_models_extra_tbl_eqtl$dataset_name}.{prediction_mashr_models_extra_tbl_eqtl$table_name} as e
    WHERE n_snps_in_model > 0
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_spredixcan_significant_genes_eqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene
  FROM ( SELECT px.gene
         FROM {predixcan_mashr_tbl_eqtl$dataset_name}.{predixcan_mashr_tbl_eqtl$table_name} as px
         JOIN {predixcan_mashr_tbl_count_eqtl$dataset_name}.{predixcan_mashr_tbl_count_eqtl$table_name} as px_count
         ON px_count.phenotype=px.phenotype
         WHERE px.pvalue < px_count.b and px.phenotype in {pheno_whitelist}
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_spredixcan_genes_eqtl <- (function(){
  query <- "
SELECT COUNT(DISTINCT gene)
FROM {predixcan_mashr_tbl_eqtl$dataset_name}.{predixcan_mashr_tbl_eqtl$table_name} 
WHERE phenotype in {pheno_whitelist}
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_dapg_genes_eqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene
  FROM ( SELECT gene
         FROM {DAPG_eqtl_clusters$dataset_name}.{DAPG_eqtl_clusters$table_name}
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_spredixcan_tuples_eqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM {predixcan_mashr_tbl_eqtl$dataset_name}.{predixcan_mashr_tbl_eqtl$table_name} as px
WHERE pvalue is not NULL and phenotype in {pheno_whitelist}
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_spredixcan_tuples_eqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM {predixcan_mashr_tbl_eqtl$dataset_name}.{predixcan_mashr_tbl_eqtl$table_name} as px
WHERE pvalue is not NULL and phenotype in {pheno_whitelist}
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_smultixcan_genes_eqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM ( 
  {multixcan_mashr_tbl_eqtl$dataset_name}.{multixcan_mashr_tbl_eqtl$table_name} as m
  INNER JOIN {multixcan_mashr_tbl_count_eqtl$dataset_name}.{multixcan_mashr_tbl_count_eqtl$table_name} as mc
  ON m.phenotype = mc.phenotype
) WHERE m.pvalue is not NULL AND mc.phenotype in {pheno_whitelist} AND m.pvalue < mc.b
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_smultixcan_tuples_eqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM {multixcan_mashr_tbl_eqtl$dataset_name}.{multixcan_mashr_tbl_eqtl$table_name} as px
WHERE pvalue is not NULL and px.phenotype in {pheno_whitelist}
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

We obtained:

- `r n_models_gt_pairs_eqtl_eur` `(gene, tissue)` pairs in MASHR Expression models
- `r n_models_genes_eqtl_eur` genes in MASHR Expression models
- `r n_models_gt_pairs_sqtl_eur` `(intron, tissue)` pairs in MASHR Expression models
- (`r n_spredixcan_significant_genes_eqtl`/`r n_spredixcan_genes_eqtl`) significant s-predixcan genes in MASHR Expression models
- `r n_spredixcan_tuples_eqtl` S-PrediXcan tuples in expression models
- `r n_smultixcan_genes_eqtl` significant s-multixcan genes in MASHR Expression models
- `r n_smultixcan_tuples_eqtl` S-MultiXcan tuples in expression models


# PrediXcan Splicing

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_models_gt_pairs_eur_sqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene, tissue
  FROM (
    SELECT gene, tissue
    FROM {prediction_mashr_models_extra_tbl_sqtl$dataset_name}.{prediction_mashr_models_extra_tbl_sqtl$table_name} as e
    WHERE n_snps_in_model > 0
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_models_gt_pairs_sqtl_eur <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene, tissue
  FROM (
    SELECT gene, tissue
    FROM {prediction_mashr_models_extra_tbl_sqtl$dataset_name}.{prediction_mashr_models_extra_tbl_sqtl$table_name} as e
    WHERE n_snps_in_model > 0
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_models_genes_sqtl_eur <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene
  FROM (
    SELECT gene
    FROM {prediction_mashr_models_extra_tbl_sqtl$dataset_name}.{prediction_mashr_models_extra_tbl_sqtl$table_name} as e
    WHERE n_snps_in_model > 0
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_spredixcan_significant_genes_sqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene
  FROM ( SELECT px.gene
         FROM {predixcan_mashr_tbl_sqtl$dataset_name}.{predixcan_mashr_tbl_sqtl$table_name} as px
         JOIN {predixcan_mashr_tbl_count_sqtl$dataset_name}.{predixcan_mashr_tbl_count_sqtl$table_name} as px_count
         ON px_count.phenotype=px.phenotype
         WHERE px.pvalue < px_count.b and px.phenotype in {pheno_whitelist}
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_spredixcan_genes_sqtl <- (function(){
  query <- "
SELECT COUNT(DISTINCT gene)
FROM {predixcan_mashr_tbl_sqtl$dataset_name}.{predixcan_mashr_tbl_sqtl$table_name} 
WHERE phenotype in {pheno_whitelist}
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_dapg_genes_sqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM (
  SELECT DISTINCT gene_id
  FROM ( SELECT gene_id
         FROM {DAPG_sqtl_clusters$dataset_name}.{DAPG_sqtl_clusters$table_name}
  )
)" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_spredixcan_tuples_sqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM {predixcan_mashr_tbl_sqtl$dataset_name}.{predixcan_mashr_tbl_sqtl$table_name} as px
WHERE pvalue is not NULL and phenotype in {pheno_whitelist}
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_spredixcan_tuples_sqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM {predixcan_mashr_tbl_sqtl$dataset_name}.{predixcan_mashr_tbl_sqtl$table_name} as px
WHERE pvalue is not NULL and phenotype in {pheno_whitelist}
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_smultixcan_tuples_sqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM {multixcan_mashr_tbl_sqtl$dataset_name}.{multixcan_mashr_tbl_sqtl$table_name} as px
WHERE pvalue is not NULL and phenotype in {pheno_whitelist}
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
n_smultixcan_genes_sqtl <- (function(){
  query <- "
SELECT COUNT(*)
FROM ( 
  {multixcan_mashr_tbl_sqtl$dataset_name}.{multixcan_mashr_tbl_sqtl$table_name} as m
  INNER JOIN {multixcan_mashr_tbl_count_sqtl$dataset_name}.{multixcan_mashr_tbl_count_sqtl$table_name} as mc
  ON m.phenotype = mc.phenotype
) WHERE m.pvalue is not NULL AND mc.phenotype in {pheno_whitelist} AND m.pvalue < mc.b
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
})() %>% as.integer
```

We obtained:

- `r n_models_gt_pairs_eur_sqtl` `(gene, tissue)` pairs in MASHR Splicing models
- `r n_models_genes_sqtl_eur` genes in MASHR Splicing models
- (`r n_spredixcan_significant_genes_sqtl`/`r n_spredixcan_genes_sqtl`) significant s-predixcan genes in MASHR Splicingmodels
- `r n_spredixcan_tuples_sqtl` S-PrediXcan tuples in splicing models
- `r n_smultixcan_genes_sqtl` significant s-multixcan genes in MASHR Splicing models
- `r n_smultixcan_tuples_sqtl` S-MultiXcan tuples in splicing models

# Tables


```{r, echo=FALSE, warning=FALSE, message=FALSE}
sp_table_ <- function(p_tbl, pc_tbl, field, field_plus){
  query <- "
SELECT p.phenotype, p.tissue, p.{field}, {field_plus} p.pvalue
FROM {p_tbl$dataset_name}.{p_tbl$table_name} as p 
INNER JOIN {pc_tbl$dataset_name}.{pc_tbl$table_name} as pc
ON p.phenotype = pc.phenotype
WHERE p.pvalue is not NULL AND p.pvalue < pc.b AND p.phenotype in {pheno_whitelist}
ORDER BY p.phenotype, p.tissue, p.pvalue
" %>% glue::glue()
  #print(query)
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
}
sp_eqtl_assoc_df <- sp_table_(predixcan_mashr_tbl_eqtl, predixcan_mashr_tbl_count_eqtl, "gene", "gene_name, ")
sp_sqtl_assoc_df <- sp_table_(predixcan_mashr_tbl_sqtl, predixcan_mashr_tbl_count_sqtl, "gene", "gene_name, ")

sm_table_ <- function(m_tbl, mc_tbl, field, field_plus){
  query <- "
SELECT m.phenotype, m.{field}, {field_plus} m.pvalue
FROM {m_tbl$dataset_name}.{m_tbl$table_name} as m
INNER JOIN {mc_tbl$dataset_name}.{mc_tbl$table_name} as mc
ON m.phenotype = mc.phenotype
WHERE m.pvalue is not NULL AND m.pvalue < mc.b AND m.phenotype in {pheno_whitelist}
ORDER BY m.phenotype, m.pvalue
" %>% glue::glue()
  #print(query)
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
}
sm_eqtl_assoc_table <- sm_table_(multixcan_mashr_tbl_eqtl, multixcan_mashr_tbl_count_eqtl, "gene", "gene_name, ")
sm_sqtl_assoc_table <- sm_table_(multixcan_mashr_tbl_sqtl, multixcan_mashr_tbl_count_sqtl, "gene", "gene_name, ")

e_table_ <- function(e_tbl){
  query <- "
SELECT phenotype, tissue, molecular_qtl_trait as gene, locus_rcp as rcp
FROM {e_tbl$dataset_name}.{e_tbl$table_name} 
WHERE locus_rcp > 0.5 AND phenotype in {pheno_whitelist}
ORDER BY phenotype, tissue, locus_rcp DESC 
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
}
sqtl_enloc_df <- e_table_(enloc_tbl_sqtl_eur)

e_table_a_ <- function(e_tbl){
  query <- "
SELECT phenotype, tissue, molecular_qtl_trait as gene, locus_rcp as rcp
FROM (
    SELECT * FROM {enloc_tbl_eqtl_eur$dataset_name}.{enloc_tbl_eqtl_eur$table_name} as e
    INNER JOIN {gencode_all_annotation_tbl$dataset_name}.{gencode_all_annotation_tbl$table_name} as a
    ON e.molecular_qtl_trait = a.gene_id
WHERE phenotype in {pheno_whitelist} AND 
      a.gene_type  in ('lincRNA', 'protein_coding', 'pseudogene') AND 
      locus_rcp > 0.5
)
 ORDER BY phenotype, tissue, locus_rcp DESC 
" %>% glue::glue()
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
}
eqtl_enloc_df <- e_table_a_(enloc_tbl_eqtl_eur)

c_table_a_ <- function(c_tbl) {
    query <- "
SELECT phenotype, tissue, gene_id as gene, pp_H4_abf as p4
FROM (
  SELECT c.phenotype, c.tissue, c.gene_id, c.pp_H4_abf FROM {c_tbl$dataset_name}.{c_tbl$table_name} as c
  INNER JOIN {gencode_all_annotation_tbl$dataset_name}.{gencode_all_annotation_tbl$table_name} as a
  ON c.gene_id = a.gene_id
  WHERE c.pp_H4_abf > 0.5 AND 
        c.phenotype in {pheno_whitelist} AND 
        a.gene_type  in ('lincRNA', 'protein_coding', 'pseudogene')
)
ORDER BY phenotype, tissue, pp_H4_abf DESC 
" %>% glue::glue()
  #print(query)
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
}
eqtl_coloc_df <- c_table_a_(coloc_tbl_eqtl )

build_sp_table <- function(sp, e) {
  sp %>%  full_join(e, by=c("phenotype", "tissue", "gene")) %>%
  mutate(is_enloc=!is.na(rcp)&rcp>0.5) %>%
  mutate(is_predixcan=!is.na(pvalue)) %>%
  select(-pvalue, -rcp)
}

eqtl_sp_df <- build_sp_table(sp_eqtl_assoc_df, eqtl_enloc_df)
sqtl_sp_df <- build_sp_table(sp_sqtl_assoc_df, sqtl_enloc_df)

build_sm_table <- function(sm, e) {
  sm %>% 
  full_join(e %>% group_by(phenotype, gene) %>% arrange(-rcp) %>% slice(1) %>% ungroup, by=c("phenotype", "gene")) %>%
  mutate(is_enloc=!is.na(rcp)&rcp>0.5) %>%
  mutate(is_multixcan=!is.na(pvalue)) %>%
  select(-pvalue, -rcp, -tissue)
}

eqtl_sm_df <- build_sm_table(sm_eqtl_assoc_table, eqtl_enloc_df)
sqtl_sm_df <- build_sm_table(sm_sqtl_assoc_table, sqtl_enloc_df)

```

## Discoveries

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#LIST OF DISCOVERIES

#S-PrediXcan
sp_table <- function(p_tbl, pc_tbl, e_tbl, field, field_plus){
  query <- "
SELECT p.phenotype, p.tissue, p.{field}, {field_plus} p.pvalue, e.locus_rcp as rcp
FROM {p_tbl$dataset_name}.{p_tbl$table_name} as p 
INNER JOIN {pc_tbl$dataset_name}.{pc_tbl$table_name} as pc
ON p.phenotype = pc.phenotype
INNER JOIN {e_tbl$dataset_name}.{e_tbl$table_name} as e
ON p.phenotype = e.phenotype AND p.tissue = e.tissue AND p.{field} = e.molecular_qtl_trait
WHERE p.pvalue is not NULL AND p.pvalue < pc.b AND e.locus_rcp > 0.5 AND p.phenotype in {pheno_whitelist}
ORDER BY p.phenotype, p.tissue, p.pvalue
" %>% glue::glue()
  #print(query)
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
}

sp_eqtl_discovery_table <- sp_table(predixcan_mashr_tbl_eqtl, predixcan_mashr_tbl_count_eqtl, enloc_tbl_eqtl_eur, "gene", "gene_name, ")
save_delim(sp_eqtl_discovery_table, fp_("spredixcan_enloc_discovery_eqtl.txt"))
#gsutil cp -a public-read output/spredixcan_enloc_discovery_eqtl.txt gs://gtex-gwas-share/data-for-paper/mashr/spredixcan_enloc_discovery_eqtl.txt

sp_sqtl_discovery_table <- sp_table(predixcan_mashr_tbl_sqtl, predixcan_mashr_tbl_count_sqtl, enloc_tbl_sqtl_eur, "gene", "gene_name, ")
sp_sqtl_discovery_table %>% rename(intron=gene) %>% select(-gene_name) %>% save_delim(fp_("spredixcan_enloc_discovery_sqtl.txt"))
#gsutil cp -a public-read output/spredixcan_enloc_discovery_sqtl.txt gs://gtex-gwas-share/data-for-paper/mashr/spredixcan_enloc_discovery_sqtl.txt

#S-MultiXcan
sm_table <- function(m_tbl, mc_tbl, e_tbl, field, field_plus){
  query <- "
SELECT m.phenotype, m.{field}, {field_plus} m.pvalue, e.locus_rcp as rcp
FROM {m_tbl$dataset_name}.{m_tbl$table_name} as m
INNER JOIN {mc_tbl$dataset_name}.{mc_tbl$table_name} as mc
ON m.phenotype = mc.phenotype
INNER JOIN (
  SELECT molecular_qtl_trait, locus_rcp, phenotype FROM (
    SELECT molecular_qtl_trait, locus_rcp, phenotype, tissue,
    ROW_NUMBER() OVER (PARTITION BY phenotype, molecular_qtl_trait ORDER BY locus_rcp DESC) as rk
    FROM {e_tbl$dataset_name}.{e_tbl$table_name} )
  WHERE locus_rcp >0.5 and rk=1) as e
ON m.phenotype = e.phenotype AND m.{field} = e.molecular_qtl_trait
WHERE m.pvalue is not NULL AND m.pvalue < mc.b AND e.locus_rcp > 0.5 AND m.phenotype in {pheno_whitelist}
ORDER BY m.phenotype, m.pvalue
" %>% glue::glue()
  #print(query)
 query_exec(query, project = "gtex-awg-im", use_legacy_sql = FALSE, max_pages = Inf)
}

sm_eqtl_discovery_table <- sm_table(multixcan_mashr_tbl_eqtl, multixcan_mashr_tbl_count_eqtl, enloc_tbl_eqtl_eur, "gene", "gene_name, ")
sm_eqtl_discovery_table %>% save_delim( fp_("smultixcan_enloc_discovery_eqtl.txt"))
#gsutil cp -a public-read output/smultixcan_enloc_discovery_eqtl.txt gs://gtex-gwas-share/data-for-paper/mashr/smultixcan_enloc_discovery_eqtl.txt

sm_sqtl_discovery_table <- sm_table(multixcan_mashr_tbl_sqtl, multixcan_mashr_tbl_count_sqtl, enloc_tbl_sqtl_eur, "gene", "gene_name, ")
sm_sqtl_discovery_table %>% rename(intron=gene) %>% select(-gene_name) %>% save_delim( fp_("smultixcan_enloc_discovery_sqtl.txt"))
#gsutil cp -a public-read output/smultixcan_enloc_discovery_sqtl.txt gs://gtex-gwas-share/data-for-paper/mashr/smultixcan_enloc_discovery_sqtl.txt

```

Enloc gene-phenotype pairs:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
data.frame(
  data = c("eqtl", "sqtl"),
  enloc = c(
    eqtl_enloc_df %>% filter(gene %in% gencode$gene_id) %>% select (phenotype, gene) %>% unique %>% nrow,
    sqtl_enloc_df %>% select (phenotype, gene) %>% unique %>% nrow
  )
) %>% kableExtra::kable(caption="gene-phenotype enloc")

```

Enloc gene-tissue pairs:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
data.frame(
  data = c("eqtl", "sqtl"),
  enloc = c(
    eqtl_enloc_df %>% select (phenotype, tissue) %>% unique %>% nrow,
    sqtl_enloc_df %>% select (phenotype, tissue) %>% unique %>% nrow
  )
) %>% kableExtra::kable(caption="tissue-phenotype enloc")

```
Enloc gene pairs:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
data.frame(
  data = c("eqtl", "sqtl"),
  enloc = c(
    eqtl_enloc_df %>% select (gene) %>% unique %>% nrow,
    sqtl_enloc_df %>% select (gene) %>% unique %>% nrow
  )
) %>% kableExtra::kable(caption="gene  enloc")

```

COLOC:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
(function(){
c_ <- eqtl_coloc_df %>% filter(gene %in% gencode$gene_id)
data.frame(
  t=c("gene-phenotype", "gene-tissue", "gene"),
  c=c(
    c_ %>% select(phenotype, gene) %>% unique %>% nrow,
    c_ %>% select(phenotype, tissue) %>% unique %>% nrow,
    c_ %>% select(gene) %>% unique %>% nrow
  )
)
})()  %>% kableExtra::kable(caption="gene  enloc")
```


Coloc versus enloc:

```{r, echo=FALSE, warning=FALSE, message=FALSE}
(function() {
  subtype_ <- function(e_, c_, gene_type_) {
    genes <- gencode %>% filter(gene_type == gene_type_) %>% .$gene_id
    data.frame( enloc= e_ %>% filter(gene %in% genes) %>% .$gene %>% unique %>%  length,
                coloc= c_ %>% filter(gene %in% genes) %>% .$gene %>% unique %>% length,
                intersection = c_ %>% inner_join(e_, by=c("phenotype", "tissue", "gene")) %>% filter(gene %in% genes) %>% .$gene %>% unique %>% length,
                total_genes = length(genes),
                gene_type = gene_type_)
  }
  rbind(
    subtype_(eqtl_enloc_df, eqtl_coloc_df, "protein_coding"),
    subtype_(eqtl_enloc_df, eqtl_coloc_df, "lincRNA")
  )
})()  %>% kableExtra::kable(caption="gene  enloc")
```

discovery gene-trait pairs:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
data.frame(
  association_method = c("s-predixcan", "s-predixcan", "s-multixcan", "s-multixcan"),
  data = c("eqtl", "sqtl", "eqtl", "sqtl"),
  discovery = c(
    eqtl_sp_df %>% filter(is_predixcan, is_enloc) %>% select(phenotype, gene) %>% unique %>% nrow,
    sqtl_sp_df %>% filter(is_predixcan, is_enloc) %>% select(phenotype, gene) %>% unique %>% nrow,
    eqtl_sm_df %>% filter(is_multixcan, is_enloc) %>% select(phenotype, gene) %>% unique %>% nrow,
    sqtl_sm_df %>% filter(is_multixcan, is_enloc) %>% select(phenotype, gene) %>% unique %>% nrow
  ),
  association = c(
    eqtl_sp_df %>% filter(is_predixcan) %>% select(phenotype, gene) %>% unique %>% nrow,
    sqtl_sp_df %>% filter(is_predixcan) %>% select(phenotype, gene) %>% unique %>% nrow,
    eqtl_sm_df %>% filter(is_multixcan) %>% select(phenotype, gene) %>% unique %>% nrow,
    sqtl_sm_df %>% filter(is_multixcan) %>% select(phenotype, gene) %>% unique %>% nrow
  ),
  enloc = c(
    eqtl_sp_df %>% filter(is_enloc) %>% select(phenotype, gene) %>% unique %>% nrow,
    sqtl_sp_df %>% filter(is_enloc) %>% select(phenotype, gene) %>% unique %>% nrow,
    eqtl_sm_df %>% filter(is_enloc) %>% select(phenotype, gene) %>% unique %>% nrow,
    sqtl_sm_df %>% filter(is_enloc) %>% select(phenotype, gene) %>% unique %>% nrow
  )
) %>% kableExtra::kable(caption="gene-phenotype discoveries")
```

discovery  tissue-trait pairs:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
data.frame(
  association_method = c("s-predixcan", "s-predixcan", "s-multixcan", "s-multixcan"),
  data = c("eqtl", "sqtl", "eqtl", "sqtl"),
  discovery = c(
    eqtl_sp_df %>% filter(is_predixcan, is_enloc) %>% select(phenotype, tissue) %>% unique %>% nrow,
    sqtl_sp_df %>% filter(is_predixcan, is_enloc) %>% select(phenotype, tissue) %>% unique %>% nrow,
    eqtl_sm_df %>% filter(is_multixcan, is_enloc) %>% select(phenotype) %>% unique %>% nrow,
    sqtl_sm_df %>% filter(is_multixcan, is_enloc) %>% select(phenotype) %>% unique %>% nrow
  ),
  association = c(
    eqtl_sp_df %>% filter(is_predixcan) %>% select(phenotype, tissue) %>% unique %>% nrow,
    sqtl_sp_df %>% filter(is_predixcan) %>% select(phenotype, tissue) %>% unique %>% nrow,
    eqtl_sm_df %>% filter(is_multixcan) %>% select(phenotype) %>% unique %>% nrow,
    sqtl_sm_df %>% filter(is_multixcan) %>% select(phenotype) %>% unique %>% nrow
  ),
  enloc = c(
    eqtl_sp_df %>% filter(is_enloc) %>% select(phenotype, tissue) %>% unique %>% nrow,
    sqtl_sp_df %>% filter(is_enloc) %>% select(phenotype, tissue) %>% unique %>% nrow,
    eqtl_sm_df %>% filter(is_enloc) %>% select(phenotype) %>% unique %>% nrow,
    sqtl_sm_df %>% filter(is_enloc) %>% select(phenotype) %>% unique %>% nrow
  )
) %>% kableExtra::kable(caption="gene-tissue discoveries")
```

